insecure/wintermute/attackOrchestrator.go

```diff
@@ -34,7 +56,259 @@ func (o *Orchestrator) WithAttackNetwork(network insecure.AttackNetwork) {
+		if receipt.ExecutionResult.ID() == o.state.originalResult.ID() {
+			// receipt contains the original result that has been corrupted.
+			// corrupted result must have already been sent to this node, so
+			// just discard it.
+			lg.Info().Msg("receipt event discarded")
+			return nil
+		}
+
+		err := o.network.Send(receiptEvent)
+		if err != nil {
+			return fmt.Errorf("could not send rpc on channel: %w", err)
+		}
+
+		lg.Info().Msg("receipt event passed through")
+		return nil
+	}
+
+	// replace honest receipt with corrupted receipt
+	corruptedResult := o.corruptExecutionResult(receipt)
+
+	corruptedExecutionIds := o.allIds.Filter(
+		filter.And(filter.HasRole(flow.RoleExecution),
+			filter.HasNodeID(o.corruptedIds...))).NodeIDs()
+
+	// sends corrupted execution result to all corrupted execution nodes.
+	for _, corruptedExecutionId := range corruptedExecutionIds {
+		// sets executor id of the result as the same corrupted execution node id that
+		// is meant to send this message to the flow network.
+		err := o.network.Send(&insecure.Event{
+			CorruptedId:       corruptedExecutionId,
+			Channel:           receiptEvent.Channel,
+			Protocol:          receiptEvent.Protocol,
+			TargetNum:         receiptEvent.TargetNum,
+			TargetIds:         receiptEvent.TargetIds,
+			FlowProtocolEvent: corruptedResult,
+		})
+		if err != nil {
+			return fmt.Errorf("could not send rpc on channel: %w", err)
+		}
+	}
+
+	// saves state of attack for further replies
+	o.state = &attackState{
+		originalResult:  &receipt.ExecutionResult,
+		corruptedResult: corruptedResult,
+	}
+	lg.Info().
+		Hex("corrupted_result_id", logging.ID(corruptedResult.ID())).
+		Msg("result successfully corrupted")
+
+	return nil
+}
+
+// handleChunkDataPackRequestEvent processes a chunk data pack request event as follows:
+// If request is for a corrupted chunk and comes from a corrupted verification node it is replied with an attestation for that
+// chunk.
+// Otherwise, it is passed through.
+func (o *Orchestrator) handleChunkDataPackRequestEvent(chunkDataPackRequestEvent *insecure.Event) error {
+	ok := o.corruptedIds.Contains(chunkDataPackRequestEvent.CorruptedId)
+	if !ok {
+		return fmt.Errorf("sender of the event is not a corrupted node")
+	}
+
+	corruptedIdentity, ok := o.allIds.ByNodeID(chunkDataPackRequestEvent.CorruptedId)
+	if !ok {
+		return fmt.Errorf("could not find corrupted identity for: %x", chunkDataPackRequestEvent.CorruptedId)
+	}
+	if corruptedIdentity.Role != flow.RoleVerification {
+		return fmt.Errorf("wrong sender role for chunk data pack request: %s", corruptedIdentity.Role.String())
+	}
+
+	if o.state != nil {
+		sent, err := o.replyWithAttestation(chunkDataPackRequestEvent)
+		if err != nil {
+			return fmt.Errorf("could not reply with attestation: %w", err)
+		}
+		if sent {
+			return nil
+		}
+	}
+
+	// no result corruption yet conducted, hence bouncing back the chunk data request.
```

Comment:
I am not sure that this statement is accurate. I think we can also reach this code _after_ some results already have been corrupted:
* the corrupted execution nodes will only corrupt a single result and then move back to honest behaviour 
* So lets look at the honest results that they emit after they corrupted their result:
  *  for the subsequent honest results, this check will be `false`: https://github.com/onflow/flow-go/blob/8c57979c888feef7e0beda782419b3723f74ee36/insecure/wintermute/attackOrchestrator.go#L275-L276 Hence, we will _not_ return here: https://github.com/onflow/flow-go/blob/8c57979c888feef7e0beda782419b3723f74ee36/insecure/wintermute/attackOrchestrator.go#L211-L213
  *  instead, we continue to https://github.com/onflow/flow-go/blob/8c57979c888feef7e0beda782419b3723f74ee36/insecure/wintermute/attackOrchestrator.go#L216-L217 despite some result already being corrupted.

In my opinion, the implementation does what we desire. Just that the documentation needs to to be revised:
```suggestion
	// chunk data request is for an honest result, hence the corrupted verifier can follow the protocol and send its honest chunk data request 
```

Reply:
Nice catch! Updated docs to reflect this clarification (couldn't commit suggestion to outdated code)

https://github.com/onflow/flow-go/pull/2291/commits/40594f7566d507a2b082b79e93de638c9fae9b7b